File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int n= strs.size();
        sort(strs.begin(),strs.end());
        if(strs.empty()) return ""; //base case
        string ans ="";
        string left = strs[0];
        string right = strs[n-1];
        int minlen = min(left.size(),right.size());
        for(int i=0;i<minlen;i++){
            if(left[i]!=right[i]) break;

            ans += left[i];
        }  
        return ans;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int n=strs.length;
        Arrays.sort(strs);
        if(n==0) return ""; //base case
        String ans="";
        String left = strs[0];
        String right = strs[n-1];
        int minlen = Math.min(left.length(),right.length());
        for(int i=0;i<minlen;i++){
            if(left.charAt(i)!=right.charAt(i)) break;
            ans+=left.charAt(i);
        }
        return ans;
    }
}